scipy/interpolate/interpolate.py

```diff
@@ -497,10 +499,6 @@ def fill_value(self):
     def fill_value(self, fill_value):
         # extrapolation only works for nearest neighbor and linear methods
         if _do_extrapolate(fill_value):
-            if self._kind not in ('nearest', 'linear'):
-                raise ValueError("Extrapolation does not work with "
-                                 "kind=%s" % self._kind)
-
```

Comment:
Because these now just works:

```
In [10]: ir = interp1d(np.arange(8), np.arange(8)**3, kind='cubic', fill_value="extrapolate")

In [11]: ir([-2, 10])
Out[11]: array([  -8., 1000.])
```